src/BluetoothGattService.cpp

```diff
@@ -117,7 +118,9 @@ BluetoothDevice BluetoothGattService::get_device ()
         throw BluetoothException(error_msg);
     }
 
-    return BluetoothDevice(device);
+    auto res = BluetoothDevice(device);
+    g_object_unref(device);
```

Comment:
I don't think these need to be freed: https://github.com/intel-iot-devkit/tinyb/blob/master/src/generated-code.c#L6595. As far as I can tell the new device doesn't have any references until BluetoothDevice gets constructed. There are a few others which are in the same situation.